typings/index.d.ts

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { FormHelperTextProps } from '@material-ui/core/FormHelperText'
+
+export interface ChipRendererArgs {
+  value: string;
+  text: string;
+  chip: string;
+  isFocused: boolean;
+  isDisabled: boolean;
+  handleClick: React.EventHandler<any>;
+  handleDelete: React.EventHandler<any>;
+  className: string;
+}
+
+export type ChipRenderer = (
+  args: ChipRendererArgs,
+  key: number
+) => React.ReactNode;
+
+export interface Props {
+  allowDuplicates?: boolean;
+  blurBehavior?: 'clear' | 'add' | 'ignore';
+  chipRenderer?: ChipRenderer;
+  clearInputValueOnChange?: boolean;
+  dataSource?: any[];
+  dataSourceConfig?: {
+    text: string;
+    value: string;
+  };
+  defaultValue?: any[];
+  disabled?: boolean;
+  FormHelperTextProps?: FormHelperTextProps;
+  fullWidth?: boolean;
+  fullWidthInput?: boolean;
+  helperText?: React.ReactNode;
+  InputLabelProps?: object;
+  InputProps?: object;
+  inputRef?: (ref: React.Ref<HTMLInputElement>) => any;
+  label?: React.ReactNode;
+  newChipKeyCodes?: number[];
+  onAdd?: (chip: string) => any;
```

Comment:
This chip could still be `any`, because a chip can either be a string or an object. Same for `onBeforeAdd` and `onChange` and `onDelete`.

Reply:
So can I type chip as string | object ?

Reply:
I'm not sure if that would break someone's project. Technically, there's nothing that prevents you from using numbers, arrays or literally *anything* as chips, as long as you provide a custom chipRenderer.

Related: https://xkcd.com/1172/

Reply:
ok doing it as any then will not hurt